transports/email/src/main/java/org/mule/transport/email/RetrieveMessageReceiver.java

```diff
@@ -62,18 +64,12 @@
     private String backupFolder = null;
     // A lock to protect concurrent access to the folder.
     private final Object folderLock = new Object();
+    // Lock to prevent connect and start from being executed simultaneously.
+    private final Object connectionLock = new Object();
 
 
     private boolean connectionEstablished = false;
```

Comment:
Change connectionEstablished datatype to AtomicBoolean since to scheduled separate threads will be reading / writing its value.

Reply:
I made the change for possible future concurrent usages of the flag, but now it is not needed since it is only used inside synchronized sections.

Reply:
Didn't noticed the read/writes where inside a synchronized section. Agree with leaving it as AtomicBoolean regardless.